### DISCUSSION TITLE
Best way to do text input masking in v7?

### DISCUSSION THREAD

**Discussion by Aryaman73** [react-hook-form/react-hook-form#6151] [Q&A] [answered]
Hey!

I stumbled upon [this video](https://www.youtube.com/watch?v=fTCTtCv8iN8) about masking using React Hook Forms, but it relies on `onChange` in the input. In v7,  I get this error: 

<img width="674" alt="Screen Shot 2021-08-02 at 7 47 06 PM" src="https://user-images.githubusercontent.com/53343807/127936933-474b2754-f305-407a-bf45-a82b14393fca.png">

 I was wondering how I could circumvent this? Or if there's a better way to mask inputs? I tried using the `pattern` attribute and though it does seem to work, it only seems to validate it on blur/submit and not as the user is typing.

**Comment by jorisre** [comment, marked as answer]
`onChange` is overwritten by `{...register('name')}` that's why you get that error.

You can solve that by:
- using `useController`
- using `Controller`
- call register in a variable: 
```tsx
const idDoFormat = register('name');

return (
  <Input
    {...idDoFormat}
    onChange={(event) => {
      idDoFormat.onChange(event);

      // Your logic
    }}
  />

)
```
